### ISSUE TITLE
what is correct range of return value from ofRandom, ofRandomf, ofRandomuf?

### ISSUE THREAD

**Issue by 2bbb** [openframeworks/openFrameworks#3762]
does openFrameworks want to `ofRandom(min, max) < max` or `ofRandom(min, max) <= max`? `ofRandomf() < 1.0f` or `ofRandomf() <= 1.0f`?

for example, in implementation of ofRandom, openFrameworks uses `rand() / (RAND_MAX + 1.0f)`.
https://github.com/openframeworks/openFrameworks/blob/master/libs/openFrameworks/math/ofMath.cpp#L48

but, i don't understand what why this code is not `rand() / static_cast<float>(RAND_MAX)`.

@novogrammer says "`RAND_MAX == 0x7fff` in Visual Studio".
then `rand() / (RAND_MAX + 1.0f) < 1.0f` is always true.
but, in Xcode/clang, `RAND_MAX == 0x7fffffff`. and this value causes underflow in float, like `static_cast<float>(RAND_MAX) == RAND_MAX + 1.0f` returns true.
and execute
```
size_t c = 0;
for(int i = 0; i &lt; RAND_MAX; i++) {
  if(1.0f == i / (RAND_MAX + 1.0f)) {
    c++;
  }
}
printf("%d", c);
```
then print `63`. this means `rand() / (RAND_MAX + 1.0f) < 1.0f` is NOT always true. sometimes `rand() / (RAND_MAX + 1.0f) == 1.0f` is true.

i think, this is bug of environment-dependent, and correct code is `rand() / (RAND_MAX + 1.0)`.

what's correct range of hope?

**Comment by bilderbuchi** [comment]
see previous (old) discussion in #224. I now think `<1`/`<max` ranges are appropriate, e.g. because there's no overlap when having multiple randoms for adjacent intevals, like (0, 1), (1, 2).

**Comment by 2bbb** [comment]
@bilderbuchi 
thanks for tell me #224, i couldn't find this issue...
i think so too. "anyone hope no overlap in multiple range."

and sorry, i found notation in newest version of `ofMath.h`:
`A random number in the range [0, max) will be returned.`
https://github.com/openframeworks/openFrameworks/blob/master/libs/openFrameworks/math/ofMath.h#L30
this is information i need...

maybe, remaining is just to solve the problem.
i will throw PR about this problem, after i do little more test.

thanks.

**Comment by bilderbuchi** [comment]
I think this can be closed now that #3842 is merged.

**Comment by 2bbb** [comment]
@bilderbuchi 
oh... sorry!!
i forgot this issue...!

thanks!!
